.github/workflows/deploy-tests.yml

```
@@ -0,0 +1,20 @@
+name: Deploy tests
+
+on:
+  pull_request:
+
+jobs:
+  assess-file-changes:
+    uses: catalystneuro/neuroconv/.github/workflows/assess-file-changes.yml@disable_tests_if_no_changes
+
+  run-tests:
+    needs: assess-file-changes
+    if: ${{ needs.assess-file-changes.outputs.SOURCE_CHANGED == 'true' }}
+    uses: catalystneuro/neuroconv/.github/workflows/testing.yml@disable_tests_if_no_changes
+    secrets:
+      DANDI_API_KEY: ${{ secrets.DANDI_API_KEY }}
+
+  run-doctest:
+    needs: assess-file-changes
+    if: ${{ needs.assess-file-changes.outputs.CONVERSION_GALLERY_CHANGED == 'true' && needs.assess-file-changes.outputs.SOURCE_CHANGED != 'true' }}
```

```suggestion
    if: ${{ needs.assess-file-changes.outputs.CONVERSION_GALLERY_CHANGED == 'true' || needs.assess-file-changes.outputs.SOURCE_CHANGED == 'true' }}
```

I think that the doctest should also be run when the source changes. One of their goals is to ensure that our documentation is up to date with our code base so we want to couple them.

---

I did that because doctest IS run as a part of the general testing workflow, no need to run it twice. The standalone workflow is just for explicitly running doctests and nothing else

---

My bad, that makes perfect sense. 
One thing we could do to avoid this confusion for someone else is to change the line:
https://github.com/catalystneuro/neuroconv/blob/a5881e98d6eb2f80e92f92ab640c32982e8ce07d/.github/workflows/testing.yml#L127-L128

To
```
        run: pytest tests/ -n auto --dist loadscope --cov=./ --cov-report xml:./codecov.xml
```

So the testing workflow files (.yml) are segregated by function (run source code tests vs run doctests). What do you think?

---

I'd probably still want it to include doctests in that single run because that's where code coverage comes from, so if there's a single line of coverage that somehow comes from the doctests but not from any other tests (kind of unlikely I suppose) we would miss it. Also they don't take very long so it doesn't take much to include them for simplicity